Makefile-constants.mk

```
@@ -1,25 +1,27 @@
-
 ifeq ($(COLLECTOR_BUILDER_TAG),)
 COLLECTOR_BUILDER_TAG=master
 endif
 
 ifeq ($(COLLECTOR_TAG),)
-ifeq ($(CIRCLE_TAG),)
 COLLECTOR_TAG=$(shell git describe --tags --abbrev=10 --dirty)
-else
-COLLECTOR_TAG := $(CIRCLE_TAG)
 endif
+
+DOCKER_CLI := $(shell command -v docker 2>/dev/null)
+
+ifeq ($(DOCKER_CLI),)
+$(error "docker is required for building")
 endif
 
+HOST_ARCH := $(shell docker system info --format '{{.Architecture}}')
+PLATFORM ?= "linux/$(HOST_ARCH)"
+
 USE_VALGRIND ?= false
 ADDRESS_SANITIZER ?= false
-CMAKE_BUILD_TYPE ?= Release
-PLATFORM ?= linux/amd64
+CMAKE_BUILD_TYPE ?= release
```

I don't think this is a major thing because CMake is case insensitive, but in all of its documentation, the build types are always camel case (`Debug`, `Release`, `RelWithDebInfo`, etc..). I think we should keep it as is for consistency at least.

https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

```suggestion
CMAKE_BUILD_TYPE ?= Release
```

---

To add to this, if the intention is to have it all lower case for consistency in the directory name I think we can do:

```
CMAKE_BASE_DIR = cmake-build-$(shell echo $(CMAKE_BUILD_TYPE) | tr A-Z a-z)-$(HOST_ARCH)
```

which is... not my favourite, but still.